release_creation/github_client/releases.py

```diff
@@ -0,0 +1,118 @@
+import copy
+import os
+import requests
+from typing import Dict, List, Optional, Set, Tuple
+from semantic_version import Version, match
+from github import Github
+from github.GithubException import GithubException
+from github.GitRelease import GitRelease
+from github.GitReleaseAsset import GitReleaseAsset
+
+_GH_SNAPSHOT_REPO = "dbt-labs/dbt-core-snapshots"
+_GH_ACCESS_TOKEN = os.environ["GH_ACCESS_TOKEN"]
+_SNAP_REQ_NAME = "snapshot_requirements"
+
+
+def get_github_client() -> Github:
+    return Github(_GH_ACCESS_TOKEN)
+
+
+def get_latest_snapshot_release(input_version: str) -> Tuple[Version, Optional[GitRelease]]:
+    gh = get_github_client()
+    target_version = Version.coerce(input_version)
+    target_match = f"~{target_version.major}.{target_version.minor}"
+    latest = copy.copy(target_version)
+    latest.patch = 0
+    repo = gh.get_repo(_GH_SNAPSHOT_REPO)
+    releases = repo.get_releases()
+    latest_release = None
+    for r in releases:
+        release_version = Version.coerce(r.tag_name)
+        if (
+            match(target_match, r.tag_name)
+            and target_version.build == release_version.build
```

Comment:
Are these fields possibly Null, as in the case of a tag without a prerelease. Maybe what's confusing me is I'm not sure what  a "build" really means here.

Reply:
indeed these fields can be null in those cases, build allows you to specify that this is for a alpha/beta/rc 
this has a few examples of what that can look like: https://pypi.org/project/semantic-version/